crates/bevy_reflect/src/serde/de.rs

```diff
@@ -538,6 +560,60 @@ impl<'a, 'de> Visitor<'de> for StructVisitor<'a> {
     }
 }
 
+struct NewtypeStructVisitor<'a> {
+    tuple_struct_info: &'static TupleStructInfo,
+    registry: &'a TypeRegistry,
+    registration: &'a TypeRegistration,
```

Comment:
What if we remove `tuple_struct_info` and then change `registration` to be `field_registration`? Then we can remove the duplicate logic and just have the caller (`TypedReflectDeserializer`/`TypedReflectSerializer`) handle getting the field registration in the newtype case.

Thoughts?

Reply:
Yeah maybe something like that. The idea is just to make it clear that we have already checked the requirements needed for it to be a true newtype. So I think that would get the point across

Reply:
I implemented this in [5e793fe](https://github.com/bevyengine/bevy/pull/8302/commits/5e793fe8897a76c8c6c33d9bbf2576ecbf24e68b) but I wonder if I should remove `NewtypeStructSerializer` entirely and inline it's logic into the calling method? Is it really important for there to be a public struct implementing that logic?

Reply:
Hm, for consistency we should probably include it. But there's not a real reason we have these all public.

I recently tried using them outside of this crate and they're actually not very helpful 😅